python/pympsim/simulator.py

```diff
@@ -20,29 +20,40 @@ def __init__(self, master):
 
         ###############John's fancy witchcraft ###################
         i = 0
+        stem_beads = 5
+        ring_beads = 60 - stem_beads
         radius = 300
         bead_radius = 10
-        stem_spacing = radius * pi * 2 / 56
-        x_offset = radius + (stem_spacing * 4) + 50
+        stem_spacing = radius * pi * 2 / ring_beads
+        stem_length = stem_spacing * stem_beads
+        x_offset = radius + stem_length + 50
         y_offset = 325
+        lords_gap_ratio = 0.2
         self.beads = []
         self.bases = []
         #make the stem
-        for i in range(0, 4):
-            x = (x_offset - radius - stem_spacing * 4) + (stem_spacing * i)
+        x = (x_offset - radius - stem_length) + (stem_spacing * i)
+        for i in range(0, 5):
             y = y_offset
             self.beads.append(self.canvas.create_oval(x, y, x+(2*bead_radius),y+(2*bead_radius), fill="#128192200", width=2))
+            x += stem_spacing
 
         #make the rest of the beads
-        for i in range(4, 60):
-            angle = (pi * 2 / 56 * (i - 4)) - pi
+        bead_spacing = (pi * 2) / (ring_beads + 1)   # there's an "extra" empty bead at the junction
+        # remove a little from bead_spacing to make room for the extra space around lords prayer beads
+        bead_spacing -= (bead_spacing * bead_radius * lords_gap_ratio) / (ring_beads + 1)
+        angle = 0 - pi + bead_spacing
+        for i in range(5, 60):
             x = x_offset + radius * cos(angle)
             y = y_offset + radius * sin(angle)
             self.beads.append(self.canvas.create_oval(x, y, x+(2*bead_radius), y+(2*bead_radius), fill="#128192200", width=2))
+            if i in [9, 10, 20, 21, 31, 32, 42, 43, 53, 54]:
+                angle += bead_spacing * lords_gap_ratio
+            angle += bead_spacing
 
         #make the bases
         for i in range(0, 9):
-            angle = (pi * 2 / 9 * (i - 4)) - pi
+            angle = (pi * 2 / 9 * (i - 6)) - pi
```

Comment:
tl;dr on the change is that the number of beads in the "stem" changed. It should actually be i - 5, not i - 6, that was a mistake.

The rest is computing the angle in radians to place each "base" triangle. Total distance around a unit circle (2pi) divided by number of bases (9) times the current base (i - 6). The we subtract pi to shift 180 degrees because of how the bases are numbered.